mods/stairs/init.lua

```diff
@@ -17,6 +16,30 @@ minetest.register_alias("stairs:slab_pinewood", "stairs:slab_pine_wood")
 
 local replace = minetest.setting_getbool("enable_stairs_replace_abm")
 
+local function rotate_and_place(itemstack, placer, pointed_thing)
+	local p0 = pointed_thing.under
+	local p1 = pointed_thing.above
+	local param2 = 0
+
+	local placer_pos = placer:getpos()
+	if placer_pos then
+		param2 = minetest.dir_to_facedir(vector.subtract(p1, placer_pos))
+	end
+
+	local finepos = minetest.pointed_thing_to_face_pos(placer, pointed_thing)
+	local _, fpos = math.modf(finepos.y)
```

Comment:
I had to look up what `math.modf` does, while `x % 1` is a pretty common idiom to get the fractional part of `x`.

Since performance does not really matter here, it's just a matter of preference.

I'm fine with it either way.